pkg/registries/types/types.go

```diff
@@ -57,3 +58,17 @@ var DockerfileInstructionSet = map[string]struct{}{
 	"VOLUME":      {},
 	"WORKDIR":     {},
 }
+
+// CreatorOptions specifies optional configuration parameters for a registry creators.
+type CreatorOptions struct {
```

Comment:
I think the options struct is fine, however I'd rather pass in optional functions in the creator function instead of a required type.

One good example of this would be [the provider options](https://github.com/stackrox/stackrox/blob/master/pkg/auth/authproviders/provider_options.go) for the auth provider creation: if you implement it this way, existing usages will stay as-is and not require changes, and only places where the option is required to be set will have changes.